### PR TITLE
feat: add mobile context communication

### PR DESCRIPTION
**Pull request by spaenleh**
This PR adds the ability for apps to work on mobile.

Things we had to change:
- The mobile app embeds apps in a WebView but does not provide the same `postMessage` APIs. We can hoverer rely on using the `window.parent.postMessage` as the mobile app will alias `window.parent` to `window.ReactNativeWebView` and the postMessage method behaves the same.
- All future communications are not made using a port but they continue using the `window.parent` as there is no real need to "isolate" from other iframes in the native context.
- add a `mobile` key to the `LocalContext` that is set to `true` when the app is embedded in mobile. This could allow the interface to adapt and display a better UI for larger touch targets. This is also used to choose the communication strategy (as stated above).

TLDR: 
1. added a new `CommunicationChannel` abstraction that allows to use the apps on mobile
2. Apps will need to be updated with this new version of the `query-client` to work on mobile

### CODE REVIEW

**Review by ReidyT — APPROVED** [review]

File: src/hooks/postMessage.ts [hunk 1]
```diff
@@ -98,6 +159,10 @@ const configurePostMessageHooks = (queryConfig: QueryClientConfig) => {
           reject(`the type '${type}' for payload '${JSON.stringify(payload)}' is not recognized`);
         }
       } catch (e) {
+        queryConfig.notifier({
+          type: 'Resolution/rejection',
```
Comment (pyphilia) [inline src/hooks/postMessage.ts:163]: Question: is it mobile specific?

**Review by pyphilia — APPROVED** [review]
LGTM 💯

File: src/hooks/postMessage.ts (hunk 1 above)
Comment (spaenleh) [reply src/hooks/postMessage.ts:163]: No, it was just missing, and I felt it would be good to send that info if it fails ?
